### DISCUSSION TITLE
How to set up IPv6 DNS Server

### DISCUSSION THREAD

**Discussion by MarthinusBosman** [0xERR0R/blocky#1240] [Q&A]
I'm not really educated enough on DNS and IPv6 stuff, but following the clear [network configuration instructions](https://0xerr0r.github.io/blocky/v0.22/network_configuration/) works well for IPv4, but my problem is that my PC seems to be using IPv6. I want to know, is it possible to do the same for IPv6, and how?

IPv6 is enabled on my router, I can see the IPv4 address of my (let's say raspberry Pi with blocky running), and can set that as my IPv4 nameserver, but for IPv6 there's 2 addresses, one `XXXX::XXXX:XXXX:XXXX:XXXX` and another `XXXX:XXXX:XXXX:0:XXXX:XXXX:XXXX:XXXX`.

Which of these do I use? I tried the first one but then everything just used the alternative nameserver, guessing it didn't work. And neither seem to be the same format as the current IPv6 nameserver.

**Comment by 0xERR0R** [comment]
You can use blocky with IPv6. First of all, please check if privacy extension is disabled (or disable it if needed :https://kalitut.com/raspberry-pi-ipv6-privacy-extensions/). With privacy extentions your raspbery pi will change the ip address periodically.

Your device will typically have multiple IPv6 addresses: public, link-local (starting with fe80) and ULA (starting with fd). You should use the ULA address in your local network. If you use docker, you have to do further steps, I assume you run blocky as binary.

Try on other machine something like `dig @fe80:xxx:xxx google.com` and you should see log output in blocky (with default settings for query log). If it works, you can configure your router, it should pass ipv4 and ipv6 addresses from your raspberry pi as DNS servers via DHCP (or configure your clients manually).

**Reply by MarthinusBosman** [reply]
Ok, so full disclosure I'm running this on TrueNAS using [TrueCharts Blocky(https://truecharts.org/charts/enterprise/blocky/setup-guide/) so it is technically in docker.

I was able to find a "fd..." IPv6 address in TrueNAS, but using `nslookup google.com` times out no matter what IPv6 address I use (I can see from the `nslookup` command that it is actually using the IPv6 address I set as the dns server.

I'm hoping there's some extra step I need to do?
